salt/cloud/clouds/digital_ocean.py

```diff
@@ -845,7 +845,12 @@ def destroy_dns_records(fqdn):
     '''
     domain = '.'.join(fqdn.split('.')[-2:])
     hostname = '.'.join(fqdn.split('.')[:-2])
-    response = query(method='domains', droplet_id=domain, command='records')
+    # TODO: remove this when the todo on 754 is available
+    try:
+        response = query(method='domains', droplet_id=domain, command='records')
+    except SaltCloudSystemExit
+        log.debug('Failed to find domains.')
```

Comment:
Oh, the reason we always attempt to delete it is because you cannot know the profile that a server was built with.  The vm_ data when deleting servers does not have any profile data, it only has the provider data, so because you can't be that granular, it is assumed that stale dns records are bad.

This is noted in the code in line 754.

```
    # TODO: when _vm config data can be made available, we should honor the configuration settings,
    # but until then, we should assume stale DNS records are bad, and default behavior should be to
    # delete them if we can. When this is resolved, also resolve the comments a couple of lines below.
```

So I would think that we should do this log and always try to delete.